src/machines/sketchSolve/sketchSolveMode.ts

```diff
@@ -1,37 +1,38 @@
-import { assign, assertEvent, createMachine, sendTo, setup } from 'xstate'
+import { assertEvent, assign, createMachine, sendTo, setup } from 'xstate'
 import type { ActorRefFrom } from 'xstate'
 import { modelingMachineDefaultContext } from '@src/machines/modelingSharedContext'
 import type {
   ModelingMachineContext,
   SetSelections,
-  MouseState,
-  SegmentOverlayPayload,
 } from '@src/machines/modelingSharedTypes'
-import type { PathToNode } from '@src/lang/wasm'
-import { machine as centerRectToolMachine } from '@src/machines/centerRectTool'
-import { machine as dimensionToolMachine } from '@src/machines/dimensionTool'
+import { machine as centerRectTool } from '@src/machines/sketchSolve/tools/centerRectTool'
+import { machine as dimensionTool } from '@src/machines/sketchSolve/tools/dimensionTool'
+import { machine as pointTool } from '@src/machines/sketchSolve/tools/pointTool'
 
-type EquipTool = 'dimension' | 'center rectangle'
+const equipTools = {
+  centerRectTool,
+  dimensionTool,
+  pointTool,
+} as const
+
+export type EquipTool = keyof typeof equipTools
 
 export type SketchSolveMachineEvent =
   | { type: 'exit' }
   | { type: 'update selection'; data?: SetSelections }
   | { type: 'unequip tool' }
   | { type: 'equip tool'; data: { tool: EquipTool } }
-  | { type: 'Set mouse state'; data: MouseState }
-  | { type: 'Set Segment Overlays'; data: SegmentOverlayPayload }
-  | { type: 'Delete segment'; data: PathToNode }
-  | { type: 'change tool'; data: { tool: EquipTool } }
-  | { type: 'click in scene'; data: [x: number, y: number] }
-  | { type: 'tool completed' }
+  | { type: 'xstate.done.actor.tool' }
 
 type ToolActorRef =
-  | ActorRefFrom<typeof dimensionToolMachine>
-  | ActorRefFrom<typeof centerRectToolMachine>
+  | ActorRefFrom<typeof dimensionTool>
+  | ActorRefFrom<typeof centerRectTool>
+  | ActorRefFrom<typeof pointTool>
```

Comment:
I'd like to see real quick if we can make this type derived so we don't need to manually add it every time.

Reply:
I think I figured out how to do this:

```ts
/**
 * using distributive conditional type
 * to wrap actor type union in ActorRefFrom
 * https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types
 */
type ToActorRefFrom<A> = A extends AnyActorLogic ? ActorRefFrom<A> : never
type ToolActorUnion = typeof equipTools[keyof typeof equipTools]
type ToolActorRef = ToActorRefFrom<ToolActorUnion>
```

But it makes me a little nervous, because we know how generics can make the XState types explode in complexity. So this is fine.